processing/src/main/java/io/druid/query/groupby/GroupByQuery.java

```diff
@@ -469,7 +473,7 @@ public Builder(GroupByQuery query)
       havingSpec = query.getHavingSpec();
       limitSpec = query.getLimitSpec();
       postProcessingFn = query.postProcessingFn;
-      context = query.getContext();
+      context = new TreeMap<>(query.getContext());
```

Comment:
I also have a similar idea, but am not sure why TreeMap is used previously. I'll fix it if there's no issues.